common/src/chain/tokens/errors.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2021 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::TokenId;
+use crate::{
+    chain::{Block, Transaction},
+    primitives::Id,
+};
+use thiserror::Error;
+
+#[derive(Error, Debug, PartialEq, Eq, Clone)]
+pub enum TokensError {
+    #[error("Incorrect ticker in issue transaction {0} in block {1}")]
+    IssueErrorIncorrectTicker(Id<Transaction>, Id<Block>),
+    #[error("Incorrect amount in issue transaction {0} in block {1}")]
+    IssueErrorIncorrectAmount(Id<Transaction>, Id<Block>),
+    #[error("Too many decimals in issue transaction {0} in block {1}")]
+    IssueErrorTooManyDecimals(Id<Transaction>, Id<Block>),
+    #[error("Incorrect metadata URI in issue transaction {0} in block {1}")]
+    IssueErrorIncorrectMetadataURI(Id<Transaction>, Id<Block>),
+    #[error("Too many tokens issued in transaction {0} in block {1}")]
+    MultipleTokenIssuanceInTransaction(Id<Transaction>, Id<Block>),
+    #[error("Coin or token overflow")]
+    CoinOrTokenOverflow,
+    #[error("Tokens fees insuffience in transaction {0} in block {1}")]
+    InsuffienceTokenFees(Id<Transaction>, Id<Block>),
+    #[error("Tokens value in inputs insuffience in transaction {0} in block {1}")]
+    InsuffienceTokenValueInInputs(Id<Transaction>, Id<Block>),
```

Comment:
InsufficientTokensInInputs

Please also fix the str descriptions

Reply:
Done! Should I change `InsuffienceTokenFees` too?

Reply:
Fixed